python/src/aiconfig/editor/server/server.py

```diff
@@ -132,15 +132,19 @@ def save() -> FlaskResponse:
     state = get_server_state(app)
     aiconfig = state.aiconfig
     request_json = request.get_json()
+    path: str | None = request_json.get("path", None)
 
-    _op = make_op_run_method(MethodName("save"))
+    if path is None:
+        if aiconfig is None:
+            return HttpResponseWithAIConfig(message="No AIConfig loaded", code=400, aiconfig=None).to_flask_format()
+        else:
+            path = aiconfig.file_path
+
+    if path is None:
+        return HttpResponseWithAIConfig(message="No path to save to", code=400, aiconfig=None).to_flask_format()
 
-    res_path_val: Result[ValidatedPath, str] = _validated_request_path(request_json, allow_create=True)
```

Comment:
Are we skipping the path validation check now? Why?

Reply:
We can't use `_validated_request_path` since it doesn't need to be in the json but thanks for pointing this out because we can/should use `get_validated_path` instead. Will update